### ISSUE TITLE
Landmark Triangulation and Local BA?

### ISSUE THREAD

**Issue by B-Meg** [ethz-asl/maplab#318]
Hey guys!
I have some questions regarding the Map Builder. How exactly does it Initialize the landmarks and how does it retriangulates them? Am I right that there is no local Bundle Adjustment, only a global one (optvi)?

**Comment by smauq** [comment]
You are right about the bundle adjustment, it's only global. As for the landmark triangulation, what in particular do you mean about how it's triangulated? The triangulation code itself is here https://github.com/ethz-asl/aslam_cv2/blob/master/aslam_cv_triangulation/src/triangulation.cc. And then multiple observations of keypoints are merged using a tracker (LK) in this case into observation tracks. Then you have a set of observations for a set of poses from vertices that you can then simply solve as a small optimization problem. Does that answer what you were asking?
